index.js

```diff
@@ -0,0 +1,100 @@
+fetch('CountriesData.json')
+  .then((response) => response.json())
+  .then((countries) => {
+    const countriesContainer = document.getElementById('countries-g');
+    if (!countriesContainer) {
+      console.error('countries-g container not found in the DOM!');
+      return;
+    }
+
+    // global variable
+    let allCountries = countries;
+
+    // view the countries
+    function displayCountries(countriesToDisplay) {
+      countriesContainer.innerHTML = ''; //cleaning all the current HTML in the div
+      countriesToDisplay.forEach((country) => {
+        const countryDiv = document.createElement('div');
+        countryDiv.className = 'country';
+
+        const countryLink = document.createElement('a');
+        countryLink.href = `details.html?country=${(country.name)}`;
+
+        countryLink.innerHTML = `
+          <div class="country-flag">
+            <img src="${country.flag}" alt="${country.name} Flag">
+          </div>
+          <div class="country-info">
+            <h2>${country.name}</h2>
+            <p><strong>Population:</strong> ${country.population}</p>
+            <p><strong>Region:</strong> ${country.region}</p>
+            <p><strong>Capital:</strong> ${country.capital}</p>
+          </div>
+        `;
+
+        countryDiv.appendChild(countryLink);
+        countriesContainer.appendChild(countryDiv);
+      });
+    }
+
+    displayCountries(allCountries);
+
+    // Filter by input
+    function filteredCountriesByInput() {
+      const filterText = document.getElementById('country-filter').value.toLowerCase();
+
+      // Filter the countries based on the input
+      const filteredCountries = allCountries.filter(country =>
+        country.name.toLowerCase().includes(filterText) || // Match for name
+        country.capital.toLowerCase().includes(filterText) || // Match for capital
```

Comment:
looks like you are iterating all over the values of the object to check if some of the values match. 

If thats the case you can do something like this : 

```js
const filteredCountries = allCountries.filter(country =>
  Object.values(country).some(value =>
    value.toString().toLowerCase().includes(filterText)
  )
);
```